apis/nucleus/src/components/listbox/ListBox.jsx

```diff
@@ -160,7 +163,7 @@ export default function ListBox({
                 const processedPages = postProcessPages ? postProcessPages(p) : p;
                 local.current.validPages = true;
                 listData.current.pages = processedPages;
-                setPages(processedPages);
+                setPages((_pages) => [...(_pages || []), ...processedPages]);
```

Comment:
Would you not keep adding to the pages in this case? What happens if you keep scrolling up and down all the time?